packages/tools/src/stateManagement/segmentation/polySeg/Surface/createAndCacheSurfacesFromRaw.ts

```diff
@@ -0,0 +1,76 @@
+import { Enums, Types, geometryLoader } from '@cornerstonejs/core';
+import { getColorForSegmentIndex } from '../../config/segmentationColor';
+import {
+  findSegmentationRepresentationByUID,
+  getSegmentation,
+} from '../../segmentationState';
+import { RawSurfacesData } from './surfaceComputationStrategies';
+import { PolySegConversionOptions } from '../../../../types';
+
+/**
+ * Creates and caches surfaces from raw surface data.
+ *
+ * @param segmentationId - The ID of the segmentation.
+ * @param rawSurfacesData - The raw surface data.
+ * @param options - Additional options for creating and caching surfaces.
+ * @param options.segmentIndices - An array of segment indices.
+ * @param options.segmentationRepresentationUID - The UID of the segmentation representation.
+ * @returns An object containing the IDs of the created surfaces.
+ */
+export async function createAndCacheSurfacesFromRaw(
+  segmentationId: string,
+  rawSurfacesData: RawSurfacesData,
+  options: PolySegConversionOptions = {}
+) {
+  // Initialize segmentationRepresentation and toolGroupId if a representation UID is provided
+  let segmentationRepresentation: any, toolGroupId: any;
+  if (options.segmentationRepresentationUID) {
+    ({ segmentationRepresentation, toolGroupId } =
+      findSegmentationRepresentationByUID(
+        options.segmentationRepresentationUID
+      ));
+  }
+
+  const segmentation = getSegmentation(segmentationId);
+
+  const geometryIds = new Map<number, string>();
+
+  // Loop through raw surfaces data and create surfaces
+  const promises = Object.keys(rawSurfacesData).map(async (index: string) => {
+    const rawSurfaceData = rawSurfacesData[index];
+    const segmentIndex = rawSurfaceData.segmentIndex;
+
+    // Get the color either from the segmentation representation or randomly generated
+    const color = segmentationRepresentation
+      ? getColorForSegmentIndex(
+          toolGroupId,
+          segmentationRepresentation.segmentationRepresentationUID,
+          segmentIndex
+        ).slice(0, 3)
+      : [Math.random() * 255, Math.random() * 255, Math.random() * 255];
```

Comment:
Does this backup happen often?  I think this should be moved into a function if it happens regularly.

Reply:
Not really, I will remove it it was for debugging